addons/web/static/src/core/select_menu/select_menu.js

```diff
@@ -38,6 +38,7 @@ export class SelectMenu extends Component {
                 shape: {
                     value: true,
                     label: { type: String },
+                    iconClass: { type: String, optional: true },
```

Comment:
Hello, it seems that it's indeed needed to add more data to the choices, but I think it would be better if you specify a prop like this `"*": true`. This way we don't need to add a new prop every time we need a new one and this would allow to more easily customize the choices based on their data.

And as it's a core component, could you also add a test in the likes of "Can add custom data to choices"? Thanks in advance!